11-Backbonejs/Backbone-Views.md

```diff
@@ -0,0 +1,321 @@
+|   `model`	|   The Backbone model which provides the data for the view.  The view's `model` can be a Backbone Model or Collection.  Each View should, in general, have **one** `model`.    	|
+|   `render`	|   A function called to draw (or redraw) the view.  By convention the render function always returns `this` so that it can be chained with other methods.  |
+
+
+## Adding our view to `app.js`
+
+We can update our renderList method to use the new view to draw each element.  Instead of calling the render method we created in `app.js` we create a new TaskView with the given model & template.
+
+Then we render the template and append the resulting `$el` to the DOM.  
+
+```javascript
+// app.js
+// imports etc...
+import TaskView from './views/task_view.js';
+
+var renderList = function(taskList) {
+  // Clear the unordered list
+  $('.todo-items').empty();
+  
+  // Iterate through the list rendering each Task
+  taskList.each(function(task) {
+  
+    // Create a new TaskView with the model & template
+    var taskView = new TaskView({
+      model: task,
+      template: _.template($('#taskItemTemplate').html()),
+      tagname: 'li'
+    });
+    
+    // Then render the TaskView
+    // And append the resulting HTML to the DOM.
+    $('.todo-items').append(taskView.render().$el);
+  });
+};
+```
+
+Notice that in the code above we used `taskView.render().$el`.  We can do this because the view's `render()` method returns a reference to the view with the line `return this;`.  It is convention in Backbone to always have the `render()` method return `this` exactly so that we can do this kind of chaining.  We also set the tagName that we are using for `el` to be `li` or a list item.
+
+We are performing many of the same operations we performed in our `app.js` file's `render` function as we create the `TaskView` and render it.
+
+## DOM Events & Views
+
+Switching to using views has rendered the delete button inoperative.  Backbone however provides a way to configure a View to respond to DOM events, like clicking on our delete button.  To do so we create an events object with the event description as the key and an event handling method.  Below we can add an events hash with our click event and an event handling method.  
+
+```javascript
+//...
+}, // render
+  events: {
+    'click button.alert': "deleteTask"
+  },
+  deleteTask: function(e) {
+    this.model.destroy();
+  }
+}};
+```
+
+
+Destroying the model triggers the "update" event in the collection and re-renders the collection.
+
+## Exercise
+
+Now try to add another event yourself toggling the task between complete or incomplete using the model method you made earlier.  You will also need to finish the event handler by calling `render()` to draw the updated Task.
+
+You can see a working version [here](https://gist.github.com/CheezItMan/ebd6a77aab299aa247ea3e9b1164dd1a).
+
+
+## Creating a View for TaskList
```

Comment:
It seems like this should be a separate resource. I see how we're adding the views for both the model and the collection, but with this set of notes that is so long, I think it would be good to try to give ourselves some logical opportunities to pause & visit learning goals before diving into more. Thoughts?